test/hotspot/jtreg/serviceability/jvmti/vthread/VThreadEventTest/libVThreadEventTest.cpp

```diff
@@ -38,10 +39,74 @@ void JNICALL VirtualThreadEnd(jvmtiEnv *jvmti, JNIEnv* jni, jthread vthread) {
 }
 
 void JNICALL VirtualThreadMount(jvmtiEnv* jvmti, ...) {
+  jvmtiError err;
+  jint count = 0;
+  jvmtiFrameInfo frameInfo[MAX_COUNT];
+
+  va_list ap;
+  JNIEnv* jni = nullptr;
+  jthread thread = nullptr;
+
+  va_start(ap, jvmti);
+  jni = va_arg(ap, JNIEnv*);
+  thread = va_arg(ap, jthread);
+  va_end(ap);
+
+  err = jvmti->GetStackTrace(thread, 0, MAX_COUNT, frameInfo, &count);
+  check_jvmti_status(jni, err, "event handler: error in JVMTI GetStackTrace call");
+
+  // Verify top 3 methods to filter events from Continuation::try_preempt().
+  const int verification_count = 3;
+  const char* expected_methods[verification_count] = {"yieldContinuation", "park", "parkVirtualThread"};
+
+  if (count < verification_count) return;
+
+  for (int idx = 0; idx < verification_count; idx++) {
+    char *methodName = nullptr;
+
+    err = jvmti->GetMethodName(frameInfo[idx].method, &methodName, nullptr, nullptr);
+    check_jvmti_status(jni, err, "event handler: error in JVMTI GetMethodName call");
+
+    if (strcmp(methodName, expected_methods[idx]) != 0) {
+      return;
+    }
+  }
   thread_mount_cnt++;
 }
 
 void JNICALL VirtualThreadUnmount(jvmtiEnv* jvmti, ...) {
+  jvmtiError err;
+  jint count = 0;
+  jvmtiFrameInfo frameInfo[MAX_COUNT];
+
+  va_list ap;
+  JNIEnv* jni = nullptr;
+  jthread thread = nullptr;
+
+  va_start(ap, jvmti);
+  jni = va_arg(ap, JNIEnv*);
+  thread = va_arg(ap, jthread);
+  va_end(ap);
+
+  err = jvmti->GetStackTrace(thread, 0, MAX_COUNT, frameInfo, &count);
+  check_jvmti_status(jni, err, "event handler: error in JVMTI GetStackTrace call");
+
+  // Verify top 3 methods to filter events from Continuation::try_preempt().
+  const int verification_count = 3;
+  const char* expected_methods[verification_count] = {"run", "enter0", "enter"};
+
+  if (count < verification_count) return;
+
+  for (int idx = 0; idx < verification_count; idx++) {
+    char *methodName = nullptr;
+
+    err = jvmti->GetMethodName(frameInfo[idx].method, &methodName, nullptr, nullptr);
+    check_jvmti_status(jni, err, "event handler: error in JVMTI GetMethodName call");
```

Comment:
Nit: There is the test library function `get_method_name()` in `jvmti_common.hpp` that can be used here.
Also, the `methodName` is better to deallocate with the `deallocate() function.
The same is in the `VirtualThreadMount` callback.

Reply:
Updated.